.circleci/config.yml

```diff
@@ -0,0 +1,71 @@
+version: 2
+jobs:
+  build_test_push:
+    docker:
+      - image: docker:stable
+    environment:
+      BUILD_IMAGE: antenna:build
+      TAGGED_IMAGE: mozilla/antenna
+      TAGGED_IMAGE_2: mozilla/socorro_collector
+    steps:
+      - checkout
+      - setup_remote_docker
+
+      - run:
+          name: Create version.json
+          command: |
+            # create a version.json per https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+            "$CIRCLE_SHA1" \
+            "$CIRCLE_TAG" \
+            "$CIRCLE_PROJECT_USERNAME" \
+            "$CIRCLE_PROJECT_REPONAME" \
+            "$CIRCLE_BUILD_URL" > version.json
+
+      - run:
+          name: Build Docker image
+          command: |
+            docker info
+            docker build -t ${BUILD_IMAGE} .
+
+      - run:
+          name: Run tests in Docker image
+          command: |
+            docker run ${BUILD_IMAGE} py.test
+            docker run ${BUILD_IMAGE} flake8
+            docker run ${BUILD_IMAGE} bandit -r antenna/
+
+      - run:
+          name: Push to Dockerhub
+          command: |
+            # images default to :latest unless there is a tag
+            if [ -n "${CIRCLE_TAG}" ]; then
+                export TAGGED_IMAGE="${TAGGED_IMAGE}:${CIRCLE_TAG}"
+                export TAGGED_IMAGE_2="${TAGGED_IMAGE_2}:${CIRCLE_TAG}"
+            fi
+
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+                echo "${DOCKER_PASSWORD}" | docker login -u="${DOCKER_USERNAME}" --password-stdin
+                docker tag "${BUILD_IMAGE}" "${TAGGED_IMAGE}"
+                docker tag "${BUILD_IMAGE}" "${TAGGED_IMAGE_2}"
+                docker push "${TAGGED_IMAGE}"
+                docker push "${TAGGED_IMAGE_2}"
```

Comment:
The `deploy-dockerhub.sh` script would retry pushing, but this doesn't. If this fails, what happens?

Reply:
If the push fails, CI will fail. This is an unfortunate side-effect. If that happens, Circle will send a failure email, and the build can be restarted.

If we were using workflows to handle this, we could restart that step independently. That creates quite a bit of complexity in the config though, which is why I avoided it. I got rid of `deploy-dockerhub.sh` for the same reason - I can add retry functionality if we see failures.